core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -82,54 +77,58 @@ public KafkaStoreReaderThread(ZkUtils zkUtils,
-    log.info("Initialized the consumer offset to " + offsetInSchemasTopic);
     Properties consumerProps = new Properties();
     consumerProps.put("group.id", this.groupId);
     consumerProps.put("client.id", "KafkaStore-reader-" + this.topic);
-    consumerProps.put("zookeeper.connect", kafkaClusterZkUrl);
-    consumerProps.put("auto.offset.reset", "smallest");
+
+    consumerProps.put("bootstrap.servers", bootstrapBrokers);
+    consumerProps.put("auto.offset.reset", "earliest");
     consumerProps.put("auto.commit.enable", "false");
-    consumer = new ZookeeperConsumerConnector(new ConsumerConfig(consumerProps));
-    Map<String, Integer> kafkaStreamConfig = new HashMap<String, Integer>();
-    kafkaStreamConfig.put(topic, 1);
-    Map<String, List<KafkaStream<byte[], byte[]>>> streams =
-        consumer.createMessageStreams(kafkaStreamConfig);
-    List<KafkaStream<byte[], byte[]>> streamsForTheLogTopic = streams.get(topic);
-    // there should be only one kafka partition and hence only one stream
-    if (streamsForTheLogTopic != null && streamsForTheLogTopic.size() != 1) {
+    consumerProps.put("key.deserializer", org.apache.kafka.common.serialization.ByteArrayDeserializer.class);
+    consumerProps.put("value.deserializer", org.apache.kafka.common.serialization.ByteArrayDeserializer.class);
+    this.consumer = new KafkaConsumer<>(consumerProps);
+
+    int partitionCount = this.consumer.partitionsFor(this.topic).size();
+    if (partitionCount < 1) {
       throw new IllegalArgumentException("Unable to subscribe to the Kafka topic " + topic +
                                          " backing this data store. Topic may not exist.");
+    } else if (partitionCount > 1) {
+      throw new IllegalStateException("Unexpected number of partitions in the " + topic +
+                                      " topic. Expected 1 and instead got " + partitionCount);
     }
-    KafkaStream<byte[], byte[]> stream = streamsForTheLogTopic.get(0);
-    consumerIterator = stream.iterator();
+
+    this.topicPartition = new TopicPartition(topic, 0);
+    this.consumer.assign(Arrays.asList(this.topicPartition));
+    this.consumer.seekToBeginning(this.topicPartition);
+
+    offsetInSchemasTopic = offsetOfLastConsumedMessage();
+    log.info("Initialized last consumed offset to " + offsetInSchemasTopic);
+
     log.debug("Kafka store reader thread started with consumer properties " +
               consumerProps.toString());
   }
 
   /**
-   * Fetch the offset of the last consumed message from ZK.
+   * Fetch the offset of the last consumed message.
    */
-  private long offsetOfLastConsumedMessage(ZkUtils zkUtils, String group, String topic) {
-    Option<String> committedOffsetStringOpt = zkUtils.readDataMaybeNull(
-        String.format("/consumers/%s/offsets/%s/0", group, topic))._1();
-    if (committedOffsetStringOpt.isEmpty()) {
+  private long offsetOfLastConsumedMessage() {
```

Comment:
If we ditch commit logic as mentioned in other comments, we don't need this method.